In [1]:
import pandas as pd

In [2]:
#reading in csv and putting into a df
purchase_df = pd.read_csv("purchase_data.csv")
purchase_df

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

In [3]:
counts = purchase_df.nunique()
counts
#SN row indicates the number of unique players
#Item ID indicates the number of unique items
#Purchase ID indicates the number of purchases

Purchase ID    780
SN             576
Age             39
Gender           3
Item ID        179
Item Name      179
Price          145
dtype: int64

In [4]:
#creating total players dataframe
total_players_df = pd.DataFrame({'Total Players': [576]})
total_players_df

Total Players
0            576

In [5]:
purchase_df.describe()
#finds average price and count of purchases


Purchase ID         Age     Item ID       Price
count   780.000000  780.000000  780.000000  780.000000
mean    389.500000   22.714103   91.755128    3.050987
std     225.310896    6.659444   52.697702    1.169549
min       0.000000    7.000000    0.000000    1.000000
25%     194.750000   20.000000   47.750000    1.980000
50%     389.500000   22.000000   92.000000    3.150000
75%     584.250000   25.000000  138.000000    4.080000
max     779.000000   45.000000  183.000000    4.990000

In [6]:
#finds total purchase value
purchase_df.sum()

Purchase ID                                               303810
SN             Lisim78Lisovynya38Ithergue48Chamassasya86Iskos...
Age                                                        17717
Gender         MaleMaleMaleMaleMaleMaleMaleMaleMaleOther / No...
Item ID                                                    71569
Item Name      Extraction, Quickblade Of Trembling HandsFrenz...
Price                                                    2379.77
dtype: object

In [7]:
#creating purchase dataframe
purchases_df = pd.DataFrame({"Number of Unique Items": [179], "Average Price": [3.05], "Number of Purchases": [780], "Total Revenue": [2379.99] })
#formatting data into dollar values
purchases_df["Average Price"] = purchases_df["Average Price"].map("${:.2f}".format)
purchases_df["Total Revenue"] = purchases_df["Total Revenue"].map("${:.2f}".format)
purchases_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780      $2379.99

In [8]:
#drops duplicates to be able to count distinct players
gender_df = purchase_df.drop_duplicates(subset='SN')

counts = gender_df['Gender'].value_counts()
#counts percentage as well formats into percentage
percentages = gender_df['Gender'].value_counts(normalize=True).mul(100).round(2).astype(str)+"%"
#create gender dataframe
mfo_df = pd.DataFrame({'Total Count': counts, 'Percentage of Players': percentages})
mfo_df




Total Count Percentage of Players
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

In [56]:
#Count of purchases by gender
count_purchases = purchase_df['Gender'].value_counts()
#Average purchase price by gender
average_price_int = purchase_df.groupby(['Gender'])
average_price_int2 = average_price_int.mean()
average_price = average_price_int2.loc[:,'Price']
#Total purchase value by gender
summed_price_int = purchase_df.groupby(['Gender'])
summed_price_int2 = summed_price_int.sum()
summed_price = summed_price_int2.loc[:,'Price']
#average total purchase per person by gender
average_total = summed_price / counts
#create summary dataframe
purchase_gender_df = pd.DataFrame({'Purchase Count': count_purchases, 'Average Purchase Price': average_price, 'Total Purchase Value': summed_price, 'Avg Total Purchase Per Person': average_total})
purchase_gender_df['Average Purchase Price'] = purchase_gender_df['Average Purchase Price'].map("${:.2f}".format)
purchase_gender_df['Total Purchase Value'] = purchase_gender_df['Total Purchase Value'].map("${:.2f}".format)
purchase_gender_df['Avg Total Purchase Per Person'] = purchase_gender_df['Avg Total Purchase Per Person'].map("${:.2f}".format)
purchase_gender_df


Purchase Count Average Purchase Price  \
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase Per Person  
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

In [60]:
#creating bins and labels
bins = [0, 9, 14, 19, 24, 29, 34, 39, 100]
group_labels = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]
gender_df['Age Bin'] = pd.cut(gender_df['Age'], bins, labels=group_labels)
gender_df.head()

#group dataframes
age_bin_df = gender_df.groupby('Age Bin')
age_bin_df.count()
age_bins_count=age_bin_df['SN'].count()
age_bins_count
age_bins_percent = (age_bins_count/age_bins_count.sum())
age_bins_percent

#create datafrmae
age_bins_final_df = pd.DataFrame({'Total Count':age_bins_count, 'Percentage of Players':age_bins_percent})
age_bins_final_df['Percentage of Players'] = age_bins_final_df['Percentage of Players'].map('{:,.2%}'.format)
age_bins_final_df

<ipython-input-60-737e7d675958>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  gender_df['Age Bin'] = pd.cut(gender_df['Age'], bins, labels=group_labels)


Total Count Percentage of Players
Age Bin                                   
<10               17                 2.95%
10-14             22                 3.82%
15-19            107                18.58%
20-24            258                44.79%
25-29             77                13.37%
30-34             52                 9.03%
35-39             31                 5.38%
40+               12                 2.08%

In [63]:
#purchase dataframe calculations
purchase_df['Age Bin'] = pd.cut(purchase_df['Age'], bins, labels=group_labels)
purchase_group_df = purchase_df.groupby('Age Bin')
purchase_bin_count = purchase_group_df['Purchase ID'].count()
purchase_bin_sum = purchase_group_df['Price'].sum()
average_purchase_price = purchase_bin_sum/purchase_bin_count
gender_df['Age Bin'] = pd.cut(gender_df['Age'], bins, labels=group_labels)
gender_df1 = gender_df.groupby('Age Bin')
distinct_person_bin = gender_df1['Purchase ID'].count()
purchase_per_person_bin = purchase_bin_sum/distinct_person_bin
purchase_per_person_bin

#creating dataframe
purchase_bin_summary = pd.DataFrame({'Purchase Count': purchase_bin_count,
                                     'Average Purchase Price': average_purchase_price,
                                    'Total Purchases': purchase_bin_sum,
                                    'Avg Total Purchase Per Person': purchase_per_person_bin})

#format the different columns 
purchase_bin_summary['Average Purchase Price'] = purchase_bin_summary['Average Purchase Price'].map("${:.2f}".format)
purchase_bin_summary['Total Purchases'] = purchase_bin_summary['Total Purchases'].map("${:.2f}".format)
purchase_bin_summary['Avg Total Purchase Per Person'] = purchase_bin_summary['Avg Total Purchase Per Person'].map("${:.2f}".format)
purchase_bin_summary


<ipython-input-63-ae4c3aca7ab7>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  gender_df['Age Bin'] = pd.cut(gender_df['Age'], bins, labels=group_labels)


Purchase Count Average Purchase Price Total Purchases  \
Age Bin                                                          
<10                  23                  $3.35          $77.13   
10-14                28                  $2.96          $82.78   
15-19               136                  $3.04         $412.89   
20-24               365                  $3.05        $1114.06   
25-29               101                  $2.90         $293.00   
30-34                73                  $2.93         $214.00   
35-39                41                  $3.60         $147.67   
40+                  13                  $2.94          $38.24   

        Avg Total Purchase Per Person  
Age Bin                                
<10                             $4.54  
10-14                           $3.76  
15-19                           $3.86  
20-24                           $4.32  
25-29                           $3.81  
30-34                           $4.12  
35-39                           $4.76  
40+                             $3.19

In [64]:
#storing variables with the necessary values
sn_df = purchase_df.groupby('SN')
sn_df1 = sn_df.count()
average_purchase_by_person = sn_df['Price'].mean()
total_purchase_by_person = sn_df['Price'].sum()
sn_df1 = sn_df1.sort_values('Purchase ID', ascending=False)
purchase_count = sn_df1['Purchase ID']
average_purchase_by_person
total_purchase_by_person
#creating the dataframe
person_purchase_df = pd.DataFrame({'Purchase Count': purchase_count,
                                  'Average Purchase Price': average_purchase_by_person,
                                  'Total Purchase Value': total_purchase_by_person})
#sorts the data with highest values at the top
person_purchase_df = person_purchase_df.sort_values('Purchase Count', ascending=False)
#format the cells
person_purchase_df['Average Purchase Price'] = person_purchase_df['Average Purchase Price'].map("${:.2f}".format)
person_purchase_df['Total Purchase Value'] = person_purchase_df['Total Purchase Value'].map("${:.2f}".format)
person_purchase_df.head()


Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Iral74                    4                  $3.40               $13.62
Idastidru52               4                  $3.86               $15.45
Asur53                    3                  $2.48                $7.44
Inguron55                 3                  $3.70               $11.11

In [80]:
#creating columns
new_df = pd.DataFrame()
new_df['Item ID'] = purchase_df['Item ID']
new_df['Item Name'] = purchase_df['Item Name']
new_df['Item Price'] = purchase_df['Price']
new_df = new_df.groupby(['Item ID', 'Item Name'])
item_price = new_df.mean()
item_price['Purchase Count'] = new_df.count()
item_price['Total Purchase Value'] = new_df.sum()
item_price = item_price.sort_values('Purchase Count', ascending=False)

#format the columns
item_price['Total Purchase Value'] = item_price['Total Purchase Value'].map("${:.2f}".format)
item_price['Item Price'] = item_price['Item Price'].map("${:.2f}".format)
item_price.head()

Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                      Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [79]:
#sort the values
profitable_items = item_price.sort_values('Total Purchase Value', ascending=False)
#format values
profitable_items['Total Purchase Value'] = profitable_items['Total Purchase Value'].map("${:.2f}".format)
profitable_items['Item Price'] = profitable_items['Item Price'].map("${:.2f}".format)
profitable_items.head()

Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                      Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

# Conclusions

1. Due to the large difference in purchase made by Males compared to Females and people who did not identify, it is clear that Males are more likely to be the purchaser of an item in the game. Males make up a large percentage of the total revenue as well because so many purchases are by males. Total purchases by Males are more than 5x the amount by Females.

2. Purchases are most likely to be made by the 20-24 year olds as they have made 365 purchases which is quite a bit more than any other age group. Additionally, they make up 44.79% of players. 

3. The user Lisosia93 has purchased the most items and dollar value. However, they do not have the highest average purchase price so they may tend to buy some less expensive items.
